pandas/tests/series/methods/test_fillna.py

```diff
@@ -623,6 +623,30 @@ def test_fillna_numeric_inplace(self):
         expected = x.fillna(value=0)
         tm.assert_series_equal(y, expected)
 
+    def test_fillna_null_value_replacement(self):
+        # GH#40498
+        ser = Series([None])
+
+        value = np.nan
+        result = ser.fillna(value)
+
+        expected = Series([np.nan])
+
+        tm.assert_series_equal(result, expected)
+
+    def test_fillna_other_missing_values_not_modified(
+        self, unique_nulls_fixture, unique_nulls_fixture2
+    ):
+        # GH#40498
+        ser = Series([1, unique_nulls_fixture, unique_nulls_fixture2, "four"])
+
+        value = {2: 0}
+        result = ser.fillna(value)
+
+        expected = Series([1, unique_nulls_fixture, 0, "four"])
+
+        tm.assert_series_equal(result, expected)
+
```

Comment:
The structure of these looks pretty similar - if possible, would be great to use the `frame_or_series` fixture to parameterize over both `frame` and `series`. By convention, the test would then live in `frame/methods`